helix-core/src/diagnostic.rs

```diff
@@ -23,11 +23,18 @@ pub struct Range {
     pub end: usize,
 }
 
+#[derive(Debug, Eq, Hash, PartialEq, Clone, Deserialize, Serialize)]
+pub enum NumberOrString {
+    Number(i32),
+    String(String),
+}
```

Comment:
Do we have to match `lsp_types` name using `NumberOrString` or would we be better off using `DiagnosticCode`?

Reply:
it can express its meaning that is good.